### ISSUE TITLE
21-Dense Text-to-Image Generation with Attention Modulation

### ISSUE THREAD

**Issue by mjmaher987** [sharif-ml-lab/visual_compositional_generation_survey#22]
## Task: Read the Paper and Do the following 4 steps
1. Highlight the paper systematically (**green** for important notes - **blue** for very important notes that worth mentioning in the final survey - **red** for the sentences that we don’t understand or need further attention in the future) [In Zotero].
2. Write the Wikipedia summary of the paper in the GitHub issue (_comment in this issue_).
3. Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
4. Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by ShayanSalehi81** [comment]
## Introduction:
This paper targeted the "Dence Caption" problems in text-to-image diffusion models, as the higher the details in the prompt go, these models show a lack of the ability to create an image without errors in objects' spatial relations, existence, and other features.

They first analyze the intermediate features of a pre-trained text-to-image diffusion model to show that the layout of a generated image is significantly related to self-attention and cross-attention maps. Based on this result they modulate intermediate attention maps according to the layout condition and further propose considering the value range of original attention scores and adjusting the degree of modulation according to the area of each segment.

As the models are typically trained on datasets with short text captions, they often fail to reflect all the details of dense captions composed of several phrases. To help to solve this problem they modulate cross-attention scores by enriching textual features for parsed texts.

## Method:
They define their condition as a set of N segments, (c_n, m_n) where each segment describes a single region. Here c_n is a non-overlapping part of the full-text caption c, and m_n denotes a binary map representing each region. This design considers the original value range and each segment’s area.

At cross-attention layers, they modulate the attention scores between paired image and text tokens in each segment (c_n, m_n) to have higher values. At self-attention layers, the modulation is applied so that pairs of image tokens belonging to the same object exhibit higher values.

They modulate the attention maps as below:

![image](https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79784127/e7cfaf2d-5128-4802-8f17-829b5d557948)

where the query-key pair condition map R defines whether to increase or decrease the attention score for a particular pair. If two tokens belong to the same segment, they form a positive pair, and their attention score will be increased. If not, they form a negative pair, with their attention decreased.

To ensure the image quality, they calculate the matrices that identify each query’s maximum and minimum values, ensuring the modulated values stay close to the original range as shown in figure below.

![image](https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79784127/c43f3281-6d79-44e6-98b9-b1b781677613)

In this paper, they observe a noticeable quality degradation when there is a large area difference between segments, to resolve this, they increase the degree for small segments and decrease for large segments.

![image](https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79784127/f834c845-553e-4c35-8d39-c3d979789870)


## Evaluation:
They evaluate each method regarding two criteria: fidelity to the text prompt and alignment with the layout condition. For text prompt, we calculate the CLIP-Score, furthermore, In terms of layout alignment, we compare IoU scores of a segmentation map predicted by YOLOv7. Also, they consider CLIP-Score to check whether generated objects followed detailed descriptions.

They also used human preference, the participants were asked to choose the better set based on either of the following criteria: fidelity to the textual condition while reflecting detailed descriptions of key objects or fidelity to the layout condition with accurate depiction of objects.

The results show that this modulation is more faithful to both textual and layout conditions. Interestingly, the performance tends to suffer significantly when the inference method is altered too much from the original one.

They also evaluate their model based on ach component used in their DenseDiffusion:
- Attention Modulation at Crossattention Layers
- Attention Modulation at Self-attention
- Value-range Adaptive Attention Modulation
- Mask-area Adaptive Attention Modulation.

from this evaluation, they understand that attention modulations in both cross-attention and self-attention layers are crucial for satisfying both textual and layout conditions. Furthermore, the value-range adaptive modulation improves the fidelity of the method to given conditions further.

The removal of each component leads to a significant drop in all metric scores. The largest performance degradation is observed when attention modulation at cross-attention layers is omitted, as textual features play a crucial role in constructing the image layout.

![image](https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79784127/707ace2e-59aa-4493-96f7-1d44763d0766)

## Fail Case:
Their proposed method fails in the latter case since the Stable Diffusion itself cannot generate an image with the correct appearance. As both self-attention and cross-attention layers are fairly insensitive, their method fails to follow fine details of the segment conditions.

## Conclusion:
The proposed method in this paper improves the fidelity of a pre-trained text-to-image model to dense captions and enables image layout control. Their finding shows that considering the value range and segment size significantly improves our attention modulation method.

It is worth mentioning that their method has two limitations.
- The method is highly dependent on the capacity of its base model, Stable Diffusion.
- The proposed method struggles with fine-grained input masks with thin structures, as both self-attention and crossattention layers are fairly coarse.
